crates/gradbench/src/cli.rs

```diff
@@ -638,17 +668,22 @@ fn cli_result() -> Result<(), ExitCode> {
                 eprintln!("error starting eval and tool commands");
                 return Err(ExitCode::FAILURE);
             };
+            let timeout = Duration::from_secs(timeout);
             let result = match output {
                 Some(path) => match fs::File::create(&path) {
-                    Ok(mut file) => intermediary(&mut file, &mut client, &mut server),
+                    Ok(mut file) => intermediary(&mut file, &mut client, &mut server, timeout),
                     Err(err) => Err(anyhow!(err)),
                 },
-                None => intermediary(&mut io::sink(), &mut client, &mut server),
+                None => intermediary(&mut io::sink(), &mut client, &mut server, timeout),
             };
             match (&result, client.wait(), server.wait()) {
-                (&Ok(invalid), Ok(e), Ok(s)) => {
+                (&Ok((invalid, timeout)), Ok(e), Ok(s)) => {
                     status_code(e)?;
-                    status_code(s)?;
+                    // In case of timeout, we don't expect the tool to
+                    // have died gracefully.
+                    if !timeout {
+                        status_code(s)?;
+                    }
```

Comment:
Can you explain the reasoning here? If the tool timed out then we should be giving a nonzero error code anyways; why special-case this?

Reply:
Never mind; somehow I missed your message above.